pkg/inventory/inventory-client.go

```diff
@@ -383,7 +393,7 @@ func (cic *UnstructuredClient) CreateOrUpdate(ctx context.Context, inv Inventory
 	if err != nil {
 		return err
 	}
-	if cic.statusPolicy != StatusPolicyAll {
+	if !cic.statusSubresource || cic.statusPolicy != StatusPolicyAll {
```

Comment:
Why do we need `statusSubresource` and `statusPolicy`? In this implementation they seem redundant

Reply:
In the current `UnstructuredClient`, `StatusPolicyAll` basically just means to call `UpdateStatus` to update the `status` subresource. That's the only thing the status policy is used for.

If the status is stored somewhere other than the `status` subresource, the `UnstructuredClient` doesn't know about it and it's an implementation detail of the conversion function

Reply:
Right. This change would allow it to accommodate the extra use case. I realized we hadn't considered it before, and it seems easy to add.

I could add a test, to prevent regressions.

Reply:
I guess what I'm saying is we don't need both of `statusPolicy` and `statusSubresource` since they both mean the same thing for the `UnstructuredClient`. They both mean that `status` is a subresource and that `UpdateStatus` needs to be called after `Update`

Reply:
`cic.toUnstructured` is called twice, once before updating the spec and once before updating the status.

How would the `ToUnstructuredFunc` implementation return once without status and once with status?

Reply:
We don't need both to support only 2 use cases, sure.
We do need both to support 3 use cases.

1. no status
2. status as subresource
3. status without subresource

Reply:
If you don't want to add the boolean, we could add more StatusPolicy enum options, like:

1. StatusPolicyNone
2. StatusPolicyEmbedded
3. StatusPolicySubresource
4. StatusPolicyField

Reply:
Alternatively, we could pass two `ToUnstructuredFunc` impls to `NewUnstructuredClient`, one that includes the status and one that doesn't. That would make implementation a little more complicated but would allow an additional optimization so that the status doesn't get computed, populated, and copied when it's not needed.

Reply:
Why does it need more enums? From the perspective of `UnstructuredClient` the only thing it needs to know is whether the status is stored in a status subresource, and if so it calls `UpdateStatus` (and does the new `status`/`spec` removal logic added by this PR).

What would the `UnstructuredClient` do with those other enum values?

Reply:
This thread is obsolete. PR was significantly refactored to remove the `statusSubresource` and `StatusPolicy` and add a `toUnstructuredStatus` instead.